packages/engine/lib/execution/src/package/simulation/context/creator.rs

```diff
@@ -16,27 +16,29 @@ pub struct ContextPackageCreators {
 }
 
 impl ContextPackageCreators {
-    pub fn from_config(_config: &PackageInitConfig) -> Result<Self> {
-        tracing::debug!("Initializing Context Package Creators");
-
-        let mut creators = HashMap::<_, Box<dyn ContextPackageCreator>>::with_capacity(3);
-        creators.insert(
-            ContextPackageName::AgentMessages,
-            Box::new(AgentMessagesCreator),
-        );
-        creators.insert(
-            ContextPackageName::ApiRequests,
-            Box::new(ApiRequestsCreator),
-        );
-        creators.insert(ContextPackageName::Neighbors, Box::new(NeighborsCreator));
-        Ok(Self { creators })
+    pub fn initialize_for_experiment_run(_config: &PackageInitConfig) -> Result<&'static Self> {
+        static PACKAGE_CREATORS: SyncOnceCell<ContextPackageCreators> = SyncOnceCell::new();
+        PACKAGE_CREATORS.get_or_try_init(|| {
+            tracing::debug!("Contextializing Context Package Creators");
```

Comment:
I can't say I understand this 😂  (not just making a joke about the typo but I don't know what you mean by Contextualising the Package Creators)

Reply:
When you replace `Init` in `Initializing` 🙃 : 927a67ede775f1cc85f0b31f31a90a1a5c6f03be